feat(Navigation): emit dedicated event for loading additional entries

blizzz opened the pull request:
## Summary

Currently, apps that add custom/dynamic entries have to listen to the `BeforeTemplateRenderedEvent` when they intend to add navigation entries. This is already rather a hack and relies on template rendering. 

But there is the endpoint of the NavigationController in core that offers to get the current data of the app navigation. The app management is making use of it, and Tables would like to do so too. But in this code path, the above mentioned Event is not being fired. Hence, fire a dedicated  event. 

This is a grey zone between a feature and a bug. For the other use case in tables we can work around it with an own endpoint, for the app management this can be considered a bug, and solving it would require the change to be backported. It would not be breaking. Since it is nowhere critical I am fine having it for 31 only.

Screenrecording of the problem:

[Screencast_20241217_202733.webm](https://github.com/user-attachments/assets/c9d76985-8255-4c34-9bd5-0ab9dca03871)

With this change and modified tables app:

[Screencast_20241217_202922.webm](https://github.com/user-attachments/assets/7886f051-2b41-4d4c-bdd5-f5b700ee7bd2)


## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [X] Screenshots before/after for front-end changes
- [ ] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [ ] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

blizzz committed:
feat(Navigation): emit dedicated event for loading additional entries

Signed-off-by: Arthur Schiwon <blizzz@arthur-schiwon.de>

provokateurin commented on lib/public/Navigation/Events/LoadAdditionalEntriesEvent.php:
```diff
@@ -0,0 +1,17 @@
+/**
+ * @since 31.0.0
+ */
+class LoadAdditionalEntriesEvent extends Event {
```
"Additional" seems a bit broad as a name. In addition to what?

blizzz replied:
the navigation entries that are loaded by the navigation manager directly

provokateurin replied:
Maybe "Custom" would be better suited? At least that feels clearer to me

blizzz replied:
What makes custom better? After all they are being additionally added.

Not that traditions are an argument by themselves, nevertheless it is more common to use `Additional` over `Custom` for events. And the words are more or less exchangeable? In our contexts at least.

```bash
$ bfs -regex '.*Additional.*Event.php'
./apps-repos/activity/lib/Event/LoadAdditionalScriptsEvent.php
./apps-repos/collectives/lib/Events/CollectivesLoadAdditionalScriptsEvent.php
./apps/files/lib/Event/LoadAdditionalScriptsEvent.php
./lib/public/Collaboration/Resources/LoadAdditionalScriptsEvent.php
./lib/public/Navigation/Events/LoadAdditionalEntriesEvent.php
$ bfs -regex '.*Load.*Event.php' 
./lib/public/BeforeSabrePubliclyLoadedEvent.php
./lib/public/Collaboration/Resources/LoadAdditionalScriptsEvent.php
./lib/public/WorkflowEngine/Events/LoadSettingsScriptsEvent.php
./lib/public/Navigation/Events/LoadAdditionalEntriesEvent.php
./apps-repos/collectives/lib/Events/CollectivesLoadAdditionalScriptsEvent.php
./apps-repos/activity/lib/Event/LoadAdditionalScriptsEvent.php
./apps/files/lib/Event/LoadAdditionalScriptsEvent.php
$ bfs -regex '.*Custom.*Event.php'
$
```

blizzz replied:
We can use Chores though – `LoadChoresEvent`  :thinking: :speak_no_evil: :face_with_peeking_eye: :innocent: :bow:

provokateurin replied:
Well pick whatever you like :woman_shrugging:

blizzz replied:
No, thank you for the suggestion! For it does not convince me, I see it as a tie between the two. If other reviewers prefer `Custom` or we find something else that fits better, I happily adjust. Atm I do not see it makes a difference.  

And then I tried to be funny and apparently failed – sorry for that.

provokateurin replied:
It wasn't unfunny :D
I just agree that any naming discussion turns out stupid anyway.

"Namen sind Schall und Rauch."

blizzz commented on lib/public/Navigation/Events/LoadAdditionalEntriesEvent.php:
```diff
@@ -0,0 +1,17 @@
+declare(strict_types=1);
+
+/**
+ * SPDX-FileCopyrightText: 2023 Nextcloud GmbH and Nextcloud contributors
```
[Off by one. What can you do.](https://twitter.com/secretGeek/status/7269997868)

blizzz commented:
Adjusted and extended test, fixed another instantiation (in file`s App.php) and updated autoloader info.

I think the failing test is unrelated, but don't really know what live photo is.

provokateurin commented:
Live photo also fails in other PRs atm, so unrelated indeed
